syntax = "proto3";
package protoconf.v1;

import "google/protobuf/any.proto"; // Moved import to the top


message ConfigSubscriptionRequest {
  string path = 1;
  string channel = 2;
  string service_name = 3;
}

message ConfigUpdate {
  google.protobuf.Any value = 1;
  string error = 2;
}

service ProtoconfService {
  rpc SubscribeForConfig(ConfigSubscriptionRequest) 
      returns (stream ConfigUpdate);
}